impeller/renderer/compute_tessellator.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include "flutter/fml/macros.h"
+#include "impeller/geometry/path.h"
+#include "impeller/renderer/buffer_view.h"
+#include "impeller/renderer/command_buffer.h"
+#include "impeller/renderer/context.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      A utility that generates triangles of the specified fill type
+///             given a polyline. This happens on the CPU.
+///
+class ComputeTessellator {
+ public:
+  ComputeTessellator();
+
+  ~ComputeTessellator();
+
+  static constexpr size_t kMaxCubicCount = 256;
+  static constexpr size_t kMaxQuadCount = 512;
+  static constexpr size_t kMaxLineCount = 1024;
+  static constexpr size_t kMaxComponentCount =
+      kMaxCubicCount + kMaxQuadCount + kMaxLineCount;
+
+  enum class Status {
+    kCommandInvalid,
+    kTooManyComponents,
+    kOk,
+  };
+
+  enum class Style {
+    kStroke,
+    // TODO(dnfield): Implement kFill.
+  };
+
+  ComputeTessellator& SetStyle(Style value);
+  ComputeTessellator& SetStrokeWidth(Scalar value);
+  ComputeTessellator& SetStrokeJoin(Join value);
+  ComputeTessellator& SetStrokeCap(Cap value);
+  ComputeTessellator& SetMiterLimit(Scalar value);
+  ComputeTessellator& SetCubicAccuracy(Scalar value);
+  ComputeTessellator& SetQuadraticTolerance(Scalar value);
+
+  //----------------------------------------------------------------------------
+  /// @brief      Generates triangles from the path.
+  ///
+  /// @return  A buffer view that will be filled with the tessellation result.
+  ///
+  Status Tessellate(
```

Comment:
How does the synchronization work here? I give you a buffer view then once the completion callback has fired then this particular compute workload is done right?